filebeat/input/log/log.go

```diff
@@ -143,19 +151,36 @@ func (f *Log) errorChecks(err error) error {
 		return ErrInactive
 	}
 
+	return nil
+}
+
+// checkFileDisappearedErrors checks if the log file has been removed or renamed (rotated).
+func (f *Log) checkFileDisappearedErrors() error {
+	// No point doing a stat call on the file if configuration options are
+	// not enabled
+	if !f.config.CloseRenamed && !f.config.CloseRemoved {
+		return nil
+	}
+
+	// Refetch fileinfo to check if the file was renamed or removed.
+	// Errors if the file was removed/rotated after reading and before
+	// calling the stat function
+	info, statErr := f.fs.Stat()
+	if statErr != nil {
+		logp.Err("Unexpected error reading from %s; error: %s", f.fs.Name(), statErr)
+		return statErr
+	}
+
 	if f.config.CloseRenamed {
 		// Check if the file can still be found under the same path
 		if !file.IsSameFile(f.fs.Name(), info) {
+			logp.Debug("harvester", "close_renamed is enabled and file %s has been renamed", f.fs.Name())
 			return ErrRenamed
 		}
 	}
 
 	if f.config.CloseRemoved {
-		// Check if the file name exists. See https://github.com/elastic/filebeat/issues/93
-		_, statErr := os.Stat(f.fs.Name())
-
-		// Error means file does not exist.
-		if statErr != nil {
+		if isRemoved(f) {
```

Comment:
@urso Per our off-PR conversation earlier today I started to remove the `os.Stat(f.fs.Name())` check here because, as you pointed out, it doesn't work correctly on Windows. I copied over your `isRemoved` implementation (see `log_other.go` and `log_windows.go` in this PR) and want to call that function from here. However, I'm not sure how to go from `f` here (which is of type `*Log`) to type `*os.File` (which is what is expected by `isRemoved`).

Reply:
Never mind, I figured it out.